ui/apps/platform/src/Containers/Vulnerabilities/VulnerablityReporting/hooks/usePageState.ts

```diff
@@ -0,0 +1,20 @@
+import useURLParameter, { QueryValue } from 'hooks/useURLParameter';
```

Comment:
From what I've seen so far. We keep track of the page state using the query params in a couple of places while we manage state in different ways:

## Collections page

We seem to be using the `useURLParamater` to set the query param to the text "`action`" and using a state and setter function to manually change the page action in the `CollectionsPage` component. We also make use of a function that checks that the actions are valid and returns some sort of prop to pass into a function called `CollectionsFormPage`. 

## Policy Management

We seem to be using the `useLocation` hook to pass the `location.search` value to a function called `parsePoliciesSearchString` that checks that the "`action`" query param is present and is a valid action using another function called `isValidAction`. If it is valid, we return the action (which converts to the type `ExtendedPageAction`). If it isn't valid, we return `undefined`. The process is very similar to Collections but done a bit differently. 

## Access Control

Like the previous two features, we also do something similar in this feature. The difference seems to be it doesn't abstract the code into separate functions. The whole process of reading from `location.search`, parsing it and converting it into an object, grabbing the `action` value, and doing checks to determine what to show, is all done in line.

## Conclusion

I think the big takeaway from all these approaches is that we need to be able to do a couple of things:
1. Read the query param `action` to get the string value
2. Validate that it is an action that we expect (like `create`, `edit`, `clone`, etc.)
3. Render appropriate components based on that value
4. Bonus: Make some way to curate paths to the right place depending on the actions. For example: If there's a button called `Create report` that takes you to the page to create a report. We need to be able to generate that link to be used in that button in a simple reusable way

This function is an attempt to be a solution for those conditions. Some highlights:
1. The generic type `T` will allow someone to use this `usePageAction` with a different set of page actions depending on the feature. So something like 

```
type ReportPageActions = 'create' | 'edit' | 'clone';
...
const { pageAction } = usePageAction<ReportPageActions>();
```
2. Makes use of the `useURLParameter` hook that Dave created to read from the `location.search` in an easier way

Reply:
If we can't do it now, we should create a Jira to update all the other places we use the final version of this new hook in the other places.

Reply:
https://issues.redhat.com/browse/ROX-18110